### PR TITLE
Add support for spec.hubTemplateOptions when generating a policy

### PR DESCRIPTION
**Pull request by mprahl**
Relates:
https://issues.redhat.com/browse/ACM-13608

### CODE REVIEW

File: internal/types/types.go [hunk 1]
```diff
@@ -7,27 +7,32 @@ import (
...
+	InformGatekeeperPolicies       bool                `json:"informGatekeeperPolicies,omitempty" yaml:"informGatekeeperPolicies,omitempty"`
+	InformKyvernoPolicies          bool                `json:"informKyvernoPolicies,omitempty" yaml:"informKyvernoPolicies,omitempty"`
+	GeneratePolicyPlacement        bool                `json:"generatePolicyPlacement,omitempty" yaml:"generatePolicyPlacement,omitempty"`
+	GeneratePlacementWhenInSet     bool                `json:"generatePlacementWhenInSet,omitempty" yaml:"generatePlacementWhenInSet,omitempty"`
+	PolicySets                     []string            `json:"policySets,omitempty" yaml:"policySets,omitempty"`
+	PolicyAnnotations              map[string]string   `json:"policyAnnotations,omitempty" yaml:"policyAnnotations,omitempty"`
+	PolicyLabels                   map[string]string   `json:"policyLabels,omitempty" yaml:"policyLabels,omitempty"`
+	ConfigurationPolicyAnnotations map[string]string   `json:"configurationPolicyAnnotations,omitempty" yaml:"configurationPolicyAnnotations,omitempty"`
+	HubTemplateOptions             *HubTemplateOptions `json:"hubTemplateOptions,omitempty" yaml:"hubTemplateOptions,omitempty"`
```
Comment (dhaiducek) [inline internal/types/types.go:35]: Looks like this pointer is copy/paste, but does it need to be a pointer when it has `omitempty`?

**Review by dhaiducek — APPROVED** [review]
/hold

LGTM! I have one question about the pointer. It seems to me the logic might be simpler without it (though it's probably fine if we're trying to be consistent with the propagator).

File: internal/types/types.go (hunk 1 above)
Comment (mprahl) [reply internal/types/types.go]: It doesn't have to be and seems to work well without it being a pointer. Updated!

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**